src/Microsoft.Health.Fhir.SqlServer/Features/Search/SqlServerSearchService.cs

```diff
@@ -128,20 +132,19 @@ private async Task<SearchResult> SearchImpl(SearchOptions searchOptions, bool hi
             // AND in the continuation token
             if (!string.IsNullOrWhiteSpace(searchOptions.ContinuationToken) && !searchOptions.CountOnly)
             {
-                if (long.TryParse(searchOptions.ContinuationToken, NumberStyles.None, CultureInfo.InvariantCulture, out var token))
+                var continuationToken = ContinuationToken.FromString(searchOptions.ContinuationToken);
```

Comment:
You've introduced a breaking change where continuation tokens from a FHIR server from a previous version will not be compatible with this version

Reply:
@johnstairs I have changed it now to support an object array as you requested.
I also left in tact the `BadRequestException` in case of invalid CT.